pkg/controller/customdomain/customdomain_controller_test.go

```diff
@@ -257,115 +117,45 @@ func TestCustomDomainController(t *testing.T) {
-		Namespace: "",
-	}, actualIngressConfig)
+		Name:      instanceName,
+		Namespace: ingressOperatorNamespace,
+	}, actualCustomIngress)
 	if err != nil {
 		t.Fatalf("get ingress: (%v)", err)
 	}
-	if actualIngressConfig.Spec.Domain != domain {
-		t.Errorf(fmt.Sprintf("CRD ingresses.config.openshift.io/cluster domain mismatch: (%v)", actualIngressConfig.Spec.Domain))
+	if actualCustomIngress.Spec.Domain != userDomain {
+		t.Errorf(fmt.Sprintf("CRD ingresscontrollers.operator.openshift.io/default domain mismatch: (%v)", actualCustomIngress.Spec.Domain))
 	}
 
-	// check annotations
+	// check instance
 	actualCustomDomain := &customdomainv1alpha1.CustomDomain{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{
-		Name:      name,
-		Namespace: namespace,
+		Name:      instanceName,
+		Namespace: instanceNamespace,
 	}, actualCustomDomain)
 	if err != nil {
 		t.Fatalf("get custom domain: (%v)", err)
 	}
-	if actualCustomDomain.ObjectMeta.Annotations == nil || actualCustomDomain.ObjectMeta.Annotations["original-domain"] != oldDomain {
-		t.Errorf(fmt.Sprintf("Problem with 'original-domain' annotation: (%+v)", actualCustomDomain.ObjectMeta.Annotations))
-	}
-	if actualCustomDomain.ObjectMeta.Annotations == nil || actualCustomDomain.ObjectMeta.Annotations["original-certificate"] != oldSecretName {
-		t.Errorf(fmt.Sprintf("Problem with 'original-certificate' annotation: (%+v)", actualCustomDomain.ObjectMeta.Annotations))
-	}
-
 	// check for ready status
 	if actualCustomDomain.Status.State != customdomainv1alpha1.CustomDomainStateReady {
 		t.Errorf(fmt.Sprintf("Status.State does not equal (%s)", string(customdomainv1alpha1.CustomDomainStateReady)))
 	}
 
-	// Check resulting modified 'system' routes
-	for n, ns := range systemRoutes {
-		actualRoute := &routev1.Route{}
-		expectedRoute := &routev1.Route{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      n,
-				Namespace: ns,
-			},
-			Spec: routev1.RouteSpec{
-				Host: n + "." + domain,
-			},
-		}
-		err = r.client.Get(context.TODO(), types.NamespacedName{
-			Name:      expectedRoute.Name,
-			Namespace: expectedRoute.Namespace,
-		}, actualRoute)
-		if err != nil {
-			t.Errorf("get customdomain: (%v)", err)
-			continue
-		}
-		if actualRoute.Name != expectedRoute.Name {
-			t.Error(fmt.Sprintf("actual route name (%v) does not match expected route name (%v)", actualRoute.Name, expectedRoute.Name))
-			continue
-		}
-		if actualRoute.Namespace != expectedRoute.Namespace {
-			t.Error(fmt.Sprintf("actual route namespace (%v) does not match expected route namespace (%v)", actualRoute.Namespace, expectedRoute.Namespace))
-			continue
-		}
-		if actualRoute.Spec.Host != expectedRoute.Spec.Host {
-			t.Error(fmt.Sprintf("actual route host (%v) does not match expected route host (%v)", actualRoute.Spec.Host, expectedRoute.Spec.Host))
-			continue
-		}
-	}
-
 	// Reconcile again so Reconcile() checks routes and updates the CustomDomain
 	// resources' Status.
 	res, err = r.Reconcile(req)
```

Comment:
I think the tail end of this test isn't doing/checking the right things. That's mostly latent, but since you're changing the objects being managed...
- After [deletion](https://github.com/openshift/custom-domains-operator/pull/11/files#diff-43760b7e5f1a3cdf576077fbc15e34c0c987b4a8a88996ed4ffb8c14d75ef743R180-R190), should check that the ingress secret and IngressController CR are gone.
- [Removing the finalizer](https://github.com/openshift/custom-domains-operator/pull/11/files#diff-43760b7e5f1a3cdf576077fbc15e34c0c987b4a8a88996ed4ffb8c14d75ef743R196-R200) is an odd thing to do. It should have been [removed by the Reconcile()](https://github.com/openshift/custom-domains-operator/pull/11/files#diff-71f68d48c4d024a88d5d71f8b173da8e4319b5e86ca6901c77e70c4b4240b4a4R115-R121). Instead check that it was removed.
- [Two short-circuit (because CustomDomain 404) Reconciles](https://github.com/openshift/custom-domains-operator/pull/11/files#diff-43760b7e5f1a3cdf576077fbc15e34c0c987b4a8a88996ed4ffb8c14d75ef743R211-R224) are well and good, but do you want to check for the absence of relevant objects after either/both?
- Could cover the Update code paths pretty easily, e.g. by changing the CustomDomain CR and Reconcile()ing again while the ingress secret and IngressController CR exist from the first run through.

Reply:
Thanks, the removal of the finalizer was just a way for me to exercise a different code path and have more code coverage. I added more checks for deleted resources towards the end of the test.